### ISSUE TITLE
`no-unsafe-assignment`: false positive on Vitest `expect.any()`

### ISSUE THREAD

**Issue by xeho91** [oxc-project/tsgolint#1096]
## Description

The `typescript/no-unsafe-assignment` rule reports a false positive when using Vitest's `expect.any(Function)` asymmetric matcher.

Vitest's `expect.any()` intentionally returns `any` by design — it's an asymmetric matcher that matches any value of the given constructor type. The rule incorrectly flags this as an unsafe assignment.

## Reproduction

```typescript
import { describe, expect, it } from "vitest";

describe("example", () => {
  it("reproduces the issue", () => {
    const result = { filepath: "...", get_json: () => ({}) };

    expect(result).toStrictEqual({
      filepath: "...",
      get_json: expect.any(Function), // ← false positive here
    });
  });
});
```

## Reported error

```
error typescript(no-unsafe-assignment): Unsafe assignment of an any value.
```

## Why this seems incorrect

The `expect.any()` matcher is a standard Vitest/Jest API that intentionally returns `any`. This is not an actual unsafe assignment — it's an asymmetric matcher used in test assertions.

The rule operates purely at the TypeScript type level using `ctx.TypeChecker.GetTypeAtLocation(senderNode)` and checking `utils.IsTypeAnyType(senderType)`. There is no special handling for Vitest/Jest matchers.

## Suggested fix

The rule should recognize `expect.any()` (and similar Vitest/Jest matchers like `expect.anything()`) as safe and not report them. This could be done by:

1. Checking if the sender node is a call expression to `expect.any` / `expect.anything`
2. Or adding an option to allow specific call expressions

## Related

- Upstream tracking issue: [oxc-project/oxc#24668](https://github.com/oxc-project/oxc/issues/24668)
- Known Vitest issue: [vitest-dev/vitest#6614](https://github.com/vitest-dev/vitest/discussions/6614)

**Comment by camc314** [comment]
Closign as not planned - please see this comment: https://github.com/vitest-dev/vitest/issues/4543#issuecomment-1824628142
